.circleci/config.yml

```diff
@@ -9,18 +9,109 @@ executors:
       xcode: 11.7
     environment:
       TERM: xterm-256color
-    
+
+aliases:
+  - &restore_macos_brew_cache
+    - when:
+          condition:
+            and:
+              - equal: [ macos-node14, << parameters.platform >> ]
+          steps:
+              - restore_cache:
+                    name: "Restore brew-gmp Cache"
+                    key: brew-gmp-cache-{{ arch }}-v1
+  - &save_macos_brew_cache
+    - when:
+          condition:
+            and:
+              - equal: [ macos-node14, << parameters.platform >> ]
+          steps:
+              - save_cache:
+                    name: "Save brew-gmp Cache"
+                    key: brew-gmp-cache-{{ arch }}-v1
+                    paths:
+                      - /usr/local/Cellar/gmp
+                      - /usr/local/lib
+                      - /usr/local/include
+
+  - &restore_cairo_cache
+    - restore_cache:
+          name: "Restore cairo-lang Cache"
+          key: cairo-lang-cache-{{ arch }}-{{ checksum "/tmp/cairo-lang-version" }}
```

Comment:
I've given this some more thought. There's a danger in doing this in case plugin's not yet been adapted to work with the latest version. This is clear if we observe the setup-venv script which has the plugin version hardcoded.

The ideal approach would be to have the current cairo-lang to which the plugin is adapted only in one place, but so far we haven't been able to come up with that.

Probably we'll have to hardcode it here as well. That way we'll have two more places (restore and save) to change when adapting to new cairo-lang versions (there are a few already across the project, e.g. Readme). Or in case we can utilize variables config.yml, there'd only be one more place.